tests/test_integration.py

```diff
@@ -1804,3 +1804,69 @@ async def test_suite_not_a_suite_directory(envsonschema, tmp_path):
         exit_code=-1,
     )
     assert re.search(r"does not contain .* cases", stderr)
+
+
+@pytest.mark.asyncio
+async def test_dialect_warning_validate(envsonschema, tmp_path):
+    tmp_path.joinpath("schema.json").write_text(
+        '{"$schema": "https://json-schema.org/draft/2020-12/schema"}',
+    )
+    tmp_path.joinpath("a.json").write_text("12")
+
+    _, stderr = await bowtie(
+        "validate",
+        "-D",
+        "7",
+        "-i",
+        envsonschema,
+        tmp_path / "schema.json",
+        tmp_path / "a.json",
+    )
+
+    assert re.search(
+        str.join(
+            r"",
+            [
+                r"\[Warning\s*\]\s*The \$Schema Poperty Refers To ",
+                r"'Draft \d{4}-\d{2}' ",
+                r"While The Dialect Argument Is 'Draft \d+'\n",
+            ],
+        ),
+        stderr.title(),
+    )
+
+
+@pytest.mark.asyncio
+async def test_dialect_warning_run(envsonschema, tmp_path):
+    tmp_path.joinpath("a.json").write_text(
+        str.join(
```

Comment:
I'm not sure why you're using `.join` here in your tests.

Have a look at the other surrounding test cases if you're trying to see how to do multiline strings in Python.

Reply:
It was due to the pre-commit errors but i guess I will just counter it with the noqa: E501 just like the other codes